plugin/pkg/admission/resourcequota/controller.go

```diff
@@ -583,6 +583,11 @@ func (e *quotaEvaluator) completeWork(ns string) {
 	e.inProgress.Delete(ns)
 }
 
+// getWork() returns a naemspace, a list of items in that namespace to
+// check, and a boolean indicating whether to shutdown.  If shutdown
+// then the list is empty.  If the list is non-empty then completeWork
```

Comment:
does the API work and is it better if we instead return the empty set of work and then the caller is always supposed to call completeWork even for an empty list?

Reply:
I am not sure I understand what "does the API work" is asking.  I tested this fix, and it seems to have banished the mysterious timeouts.

I considered the alternative you suggested.  It makes for code that is simpler but less efficient (it costs an extra mutex entry) in the case when the list is empty.  I suspect the empty list case is rare, so I would be happy to simplify the code if you guys would prefer that.

Reply:
> I considered the alternative you suggested. It makes for code that is simpler but less efficient (it costs an extra mutex entry) in the case when the list is empty. I suspect the empty list case is rare, so I would be happy to simplify the code if you guys would prefer that.

Yes, please.  Let's do that.  I think it will help future bug fixers figure out what's happening.